apps/api/src/service/application-service.ts

```diff
@@ -0,0 +1,164 @@
+				application_id: id,
+				created_at: new Date(),
+				updated_at: new Date(),
+			};
+			const newAppContentsRecord = await db.insert(applicationContents).values(newAppContents).returning();
+			const { id: contentsId } = newAppContentsRecord[0];
+
+			const application = await db
+				.update(applications)
+				.set({ contents: contentsId })
+				.where(eq(applications.id, id))
+				.returning();
+
+			console.log(`Application created with user_id: ${user_id}`);
+
+			return application[0];
+		} catch (err) {
+			console.error(`Error at createApplication with user_id: ${user_id}`);
+			console.error(err);
+			return null;
+		}
+	},
+	findOneAndUpdate: async ({ id, update }: { id: number; update: any }) => {
+		try {
+			const application = await db
+				.update(applications)
+				.set({ ...update, updated_at: sql`NOW()` })
+				.where(eq(applications.id, id))
+				.returning();
+
+			return application;
+		} catch (err) {
+			console.error(`Error at findOneAndUpdate with id: ${id}`);
+			console.error(err);
+			return null;
+		}
+	},
+	getApplicationById: async ({ id }: { id: number }) => {
+		try {
+			const applicationRecord = await db
+				.select()
+				.from(applications)
+				.where(eq(applications.id, id))
+				.leftJoin(applicationContents, eq(applications.contents, applicationContents.id));
+
+			const application = {
+				...applicationRecord[0].applications,
+				contents: applicationRecord[0].application_contents,
+			};
+
+			return application;
+		} catch (err) {
+			console.error(`Error at getApplicationById with id: ${id}`);
+			console.error(err);
+			return null;
+		}
+	},
+	listApplications: async ({
+		user_id,
+		state,
+		sort = '',
+	}: {
+		user_id?: string;
+		state?: ApplicationStates;
+		sort?: string;
+	}) => {
+		const isDescending = sort?.charAt(0) === '-';
+		const sortValue = isDescending ? sort.substring(1) : sort;
+		const sortFunction = isDescending ? desc : asc;
+
+		let sortKey;
+
+		switch (sortValue) {
+			case 'updated_at':
+				sortKey = applications.updated_at;
+				break;
+			case 'state':
+				sortKey = applications.state;
+				break;
+			case 'created_at':
+			default:
+				sortKey = applications.created_at;
+				break;
+		}
+
+		const sortQuery = sortFunction(sortKey);
+
+		try {
+			const allApplications = await db
+				.select({
+					id: applications.id,
+					user_id: applications.user_id,
+					state: applications.state,
+					createdAt: applications.created_at,
+					updatedAt: applications.updated_at,
+				})
+				.from(applications)
+				.where(
+					and(
+						user_id ? eq(applications.user_id, String(user_id)) : undefined,
+						state ? eq(applications.state, state) : undefined,
+					),
+				)
+				.orderBy(sortQuery);
+
+			return allApplications;
+		} catch (err) {
+			console.error(`Error at listApplications with user_id: ${user_id} state: ${state}`);
+			console.error(err);
+			return null;
+		}
+	},
+	deleteApplication: async ({ user_id }: { user_id: string }) => {
```

Comment:
Similar to `findOneAndUpdate`, there was a need to delete records after they were created.

There were connection issues (now fixed) creating two separate DB connections (one for the service, one for the tests) so I needed additional service methods first to resolve the test issues before I diagnosed the connection issues.

Reply:
We have no use case for deleting applications. In order to prevent this being used when we shouldnt, can we remove this function?

Reply:
Removed